src/main/java/com/surrealdb/config/SurrealCrudRepositoryImpl.java

```diff
@@ -0,0 +1,142 @@
+package com.surrealdb.config;
+
+import com.surrealdb.driver.SyncSurrealDriver;
+import com.surrealdb.driver.model.QueryResult;
+import com.surrealdb.refactor.exception.MissingSurrealTableAnnotationException;
+
+import java.lang.reflect.Field;
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+import static com.surrealdb.config.SurrealDBConnection.getRepoDriver;
+
+public class SurrealCrudRepositoryImpl<T, ID> implements SurrealCrudRepository<T, ID> {
+
+    private final Class<T> entityType;
+    SyncSurrealDriver driver = getRepoDriver();
+    private final String tableName;
+
+    public SurrealCrudRepositoryImpl(Class<T> entityType) {
+        this.entityType = entityType;
+        this.tableName = resolveTableName(entityType);
+    }
+
+    private static String resolveTableName(Class<?> entityType) {
+        if (!entityType.isAnnotationPresent(SurrealTable.class)) {
+            throw new MissingSurrealTableAnnotationException(entityType);
+        }
+        return entityType.getAnnotation(SurrealTable.class).value();
+    }
+
+    @Override
+    public Optional<T> findById(ID id) {
+        String query = String.format("select * from %s where id = '%s';", tableName, id.toString());
```

Comment:
This is a security vulnerability - see "sql injection". It happens when you take input and put it into a trusted query. You should instead use the protocol-level parameter handling.